src/coreclr/gc/gc.cpp

```diff
@@ -33324,22 +33324,20 @@ void gc_heap::walk_relocation (void* profiling_context, record_surv_fn fn)
     int condemned_gen_number = settings.condemned_generation;
     int stop_gen_idx = get_stop_generation_index (condemned_gen_number);
 
+    reset_pinned_queue_bos();
+    update_oldest_pinned_plug();
+
     for (int i = condemned_gen_number; i >= stop_gen_idx; i--)
     {
         generation* condemned_gen = generation_of (i);
-        heap_segment*  current_heap_segment = heap_segment_rw (generation_start_segment (condemned_gen));
-#ifdef USE_REGIONS
-        current_heap_segment = walk_relocation_sip (current_heap_segment, profiling_context, fn);
+        heap_segment*  current_heap_segment = generation_start_segment (condemned_gen);
```

Comment:
I presume you will revert these changes (except keep the `reset_pinned_queue_bos/update_oldest_pinned_plug` part)?